### PR TITLE
Remove EditorFeatures.Cocoa and VSMac IVTs

### LINKED ISSUE #35099: Migrate Xamarin IVTs to the External Access pattern
We are planning to remove IVT access to dotnet/roslyn from the following assembly for 16.2 Preview 1:

* Xamarin.Ide

This assembly should be updated to operate correctly without IVT access to roslyn, or use an External Access assembly for all access to Roslyn internals. I will be assisting in the creation of the External Access layer if necessary for these components.

### PR DESCRIPTION
**Pull request by sandyarmstrong**
VSMac only needs to be supported on the 17.6 branch now.

Closes #35099

### CODE REVIEW

**Review by davidwengier — APPROVED** [review]
🫡

**Review by jasonmalinowski — APPROVED** [review]
😢

**Comment by sandyarmstrong** [comment]
I don't _think_ my changes could have caused this failure, and the test appears to be passing locally, but I can't retrigger jobs so I'm not sure.

> [xUnit.net 00:00:54.92]     Microsoft.CodeAnalysis.Editor.UnitTests.RenameTracking.RenameTrackingTaggerProviderTests.RenameTrackingSessionContinuesAfterViewingTag [FAIL]
  Failed Microsoft.CodeAnalysis.Editor.UnitTests.RenameTracking.RenameTrackingTaggerProviderTests.RenameTrackingSessionContinuesAfterViewingTag [48 ms]
  Error Message:
   Assert.Equal() Failure
Expected: 1
Actual:   0
  Stack Trace:
     at Microsoft.CodeAnalysis.Editor.UnitTests.RenameTracking.RenameTrackingTestState.<AssertTag>d__27.MoveNext() in /_/src/EditorFeatures/Test/RenameTracking/RenameTrackingTestState.cs:line 190

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -20,11 +20,8 @@ src/Compilers/ @dotnet/roslyn-compiler
 src/Compilers/Test/Core/Traits/Traits.cs
 src/EditorFeatures/ @dotnet/roslyn-ide
 
-# If we touch the WPF layer or Cocoa layers, the VS for Mac team also should take a look. We do this even if it's
-# only touching WPF since the change might also need to be made (and validated) on the Cocoa side of things.
-src/EditorFeatures/Core.Cocoa/ @dotnet/roslyn-ide @dotnet/roslyn-vs-for-mac
-src/EditorFeatures/Core.Wpf/ @dotnet/roslyn-ide @dotnet/roslyn-vs-for-mac
-src/EditorFeatures/CSharp.Wpf/ @dotnet/roslyn-ide @dotnet/roslyn-vs-for-mac
+src/EditorFeatures/Core.Wpf/ @dotnet/roslyn-ide
+src/EditorFeatures/CSharp.Wpf/ @dotnet/roslyn-ide
```
Comment (sharwell) [inline .github/CODEOWNERS:24]: These are covered by the line above and can be removed

**Comment by jasonmalinowski** [comment]
@sandyarmstrong Not likely, I'll retrigger them.

File: src/EditorFeatures/CSharp/Microsoft.CodeAnalysis.CSharp.EditorFeatures.csproj [hunk 2]
```diff
@@ -33,19 +33,6 @@
     <InternalsVisibleTo Include="Microsoft.CodeAnalysis.Workspaces.Test.Utilities" />
     <InternalsVisibleTo Include="Microsoft.CodeAnalysis.EditorFeatures.Test.Utilities2" />
     <InternalsVisibleTo Include="Microsoft.VisualStudio.LanguageServices.LiveShare" />
-    <!-- BEGIN MONODEVELOP
-    These MonoDevelop dependencies don't ship with Visual Studio, so can't break our
-    binary insertions and are exempted from the ExternalAccess adapter assembly policies.
```
Comment (sharwell) [inline src/EditorFeatures/CSharp/Microsoft.CodeAnalysis.CSharp.EditorFeatures.csproj:38]: 🥲

File: src/EditorFeatures/Core.Wpf/InlineRename/CommandHandlers/RenameCommandHandler.cs [hunk 3]
```diff
@@ -14,9 +14,7 @@
 using Microsoft.CodeAnalysis.Telemetry;
 using Microsoft.CodeAnalysis.Shared.TestHooks;
 
-#if !COCOA
 using System.Linq;
```
Comment (sharwell) [inline src/EditorFeatures/Core.Wpf/InlineRename/CommandHandlers/RenameCommandHandler.cs:17]: 💡 Can be sorted into the list above

**Review by sharwell — APPROVED** [review]

**Comment by sandyarmstrong** [comment]
@jasonmalinowski looks good after latest push, can I go ahead and merge?

**Comment by jasonmalinowski** [comment]
@sandyarmstrong Yep, you're good.

**Pull request merged**